zhmcclient/_adapter.py

```diff
@@ -0,0 +1,178 @@
+        """
+        super(AdapterManager, self).__init__(cpc)
+
+    @property
+    def cpc(self):
+        """
+        :class:`~zhmcclient.Cpc`: Parent object (CPC) defining the scope for
+        this manager object.
+        """
+        return self._parent
+
+    def list(self, full_properties=False):
+        """
+        List the adapters in scope of this manager object.
+
+        Parameters:
+
+          full_properties (bool):
+            Controls whether the full set of resource properties should be
+            retrieved, vs. only the short set as returned by the list
+            operation.
+
+        Returns:
+
+          : A list of :class:`~zhmcclient.Adapter` objects.
+
+        Raises:
+
+          :exc:`~zhmcclient.HTTPError`
+          :exc:`~zhmcclient.ParseError`
+          :exc:`~zhmcclient.AuthError`
+          :exc:`~zhmcclient.ConnectionError`
+        """
+        cpc_uri = self.cpc.get_property('object-uri')
+        adapters_res = self.session.get(cpc_uri + '/adapters')
+        adapter_list = []
+        if adapters_res:
+            adapter_items = adapters_res['adapters']
+            for adapter_props in adapter_items:
+                adapter = Adapter(self, adapter_props['object-uri'],
+                                  adapter_props)
+                if full_properties:
+                    adapter.pull_full_properties()
+                adapter_list.append(adapter)
+        return adapter_list
+
+    def create_hipersocket(self, properties):
+        """
+        Create and configures a HiperSockets adapter
+        with the specified resource properties.
+
+        Parameters:
+
+          properties (dict): Properties for the new adapter.
+
+        Returns:
+
+          string: The resource URI of the new adapter.
+
+        Raises:
+
+          :exc:`~zhmcclient.HTTPError`
+          :exc:`~zhmcclient.ParseError`
+          :exc:`~zhmcclient.AuthError`
+          :exc:`~zhmcclient.ConnectionError`
+        """
+        cpc_uri = self.cpc.get_property('object-uri')
+        result = self.session.post(cpc_uri + '/adapters', body=properties)
+        return result['object-uri']
+
+
+class Adapter(BaseResource):
+    """
+    Representation of an Adapter.
+
+    Derived from :class:`~zhmcclient.BaseResource`; see there for common
+    methods and attributes.
+    """
+
+    def __init__(self, manager, uri, properties):
+        """
+        Parameters:
+
+          manager (:class:`~zhmcclient.AdapterManager`):
+            Manager object for this resource.
+
+          uri (string):
+            Canonical URI path of the Adapter object.
+
+          properties (dict):
+            Properties to be set for this resource object.
+            See initialization of :class:`~zhmcclient.BaseResource` for
+            details.
+        """
+        assert isinstance(manager, AdapterManager)
+        super(Adapter, self).__init__(manager, uri, properties)
+
+    def delete_hipersocket(self):
```

Comment:
Why is a HiperSockets adapter any special when deleting it? Of course, it is currently the only type that can be deleted, but if a second type comes up tomorrow that can be deleted, I don't think that we would like to add a second `delete_xxx()` method. I suggest to rename this method to just `delete()`, and let the HMC decide whether the URI can be deleted. The description may very well state that only HiperSockets adapters can be deleted, and that deletion attempts for other adapters will be rejected.